include/seqan3/std/span

```diff
@@ -19,9 +19,11 @@
 
 #if __has_include(<span>)
 #include <span>
+#endif
```

Comment:
what if __has_include(<span>) is true but cpp_lib_span isn't? Can we guarantee that this never happens?

Reply:
The header span is always available >= gcc10. cpp_lib_span will only be defined if you build in c++2a mode.